app/controllers/providers/allocations_controller.rb

```diff
@@ -3,11 +3,14 @@ class AllocationsController < ApplicationController
     before_action :build_allocations
     before_action :build_recruitment_cycle
     before_action :build_provider
+    before_action :build_training_provider
```

Comment:
```suggestion
    before_action :build_training_provider, except: %i[index]
```
maybe we should start restricting those before methods with `only` / `except`